src/runtime/entitlements-manager.js

```diff
@@ -108,12 +119,25 @@ export class EntitlementsManager {
   }
 
   /**
-   * @param {?string=} encryptedDocumentKey
+   * @param {!GetEntitlementsParams=} params
    * @return {!Promise<!Entitlements>}
    */
-  getEntitlements(encryptedDocumentKey) {
+  getEntitlements(params) {
+    // Remain backwards compatible by accepting
+    // `encryptedDocumentKey` string as a first param.
+    if (typeof params === 'string') {
+      // TODO: Delete the fallback if nobody needs it. Use a log to verify.
+      warn(
+        `[swg.js:getEntitlements]: The first param of getEntitlements() should be an object of type GetEntitlementsParams.`
```

Comment:
... if present ...

Reply:
Maybe comment out the warn until we have the matching AMP code deployed so as not to confuse people..

Reply:
I added a `Date.now()` which will block this warning for 2 weeks